tests/js-modules/modules.py

```diff
@@ -948,6 +947,99 @@ def test_npm_app(network, args):
+                },
+            )
+            assert r.status_code != http.HTTPStatus.OK, r.status_code
+
+            # Test too short a quote
+            r = c.post(
+                "/app/verifySnpAttestation",
+                {
+                    "evidence": primary_quote_info["raw"][:-10],
+                    "endorsements": primary_quote_info["endorsements"],
+                    "endorsed_tcb": primary_quote_info["endorsed_tcb"],
+                },
+            )
+            assert r.status_code != http.HTTPStatus.OK, r.status_code
+
+            # Test too long a quote
+            r = c.post(
+                "/app/verifySnpAttestation",
+                {
+                    "evidence": primary_quote_info["raw"] + "1",
+                    "endorsements": primary_quote_info["endorsements"],
+                    "endorsed_tcb": primary_quote_info["endorsed_tcb"],
+                },
+            )
+            assert r.status_code != http.HTTPStatus.OK, r.status_code
+
+            # Test too short an endorsement
+            r = c.post(
+                "/app/verifySnpAttestation",
+                {
+                    "evidence": primary_quote_info["raw"],
+                    "endorsements": primary_quote_info["endorsements"][:-10],
+                    "endorsed_tcb": primary_quote_info["endorsed_tcb"],
+                },
+            )
+            assert r.status_code != http.HTTPStatus.OK, r.status_code
+
+            # Test too long an endorsement
+            r = c.post(
+                "/app/verifySnpAttestation",
+                {
+                    "evidence": primary_quote_info["raw"],
+                    "endorsements": primary_quote_info["endorsements"] + "1",
+                    "endorsed_tcb": primary_quote_info["endorsed_tcb"],
+                },
+            )
+            assert r.status_code != http.HTTPStatus.OK, r.status_code
+
+            def corrupt_value(value: str):
+                return value[len(value) // 2 + 1 :] + value[: len(value) // 2]
+
+            # Test corrupted quote
+            r = c.post(
+                "/app/verifySnpAttestation",
+                {
+                    "evidence": corrupt_value(primary_quote_info["raw"]),
+                    "endorsements": primary_quote_info["endorsements"],
+                    "endorsed_tcb": primary_quote_info["endorsed_tcb"],
+                },
+            )
+            assert r.status_code != http.HTTPStatus.OK, r.status_code
+
+            # Test corrupted endorsements
+            r = c.post(
+                "/app/verifySnpAttestation",
+                {
+                    "evidence": primary_quote_info["raw"],
+                    "endorsements": corrupt_value(primary_quote_info["endorsements"]),
+                    "endorsed_tcb": primary_quote_info["endorsed_tcb"],
+                },
+            )
+            assert r.status_code != http.HTTPStatus.OK, r.status_code
```

Comment:
This is not good practice, if the code throws and returns a 500, this may pass for the wrong reason. We should check for the expected error code here (4xx probably?)

Reply:
https://github.com/microsoft/CCF/pull/5653/commits/f5c4991f3d961435a114422a21a47d318c9a8be7